core/src/test/java/nl/nn/adapterframework/webcontrol/api/ApiTestBase.java

```diff
@@ -71,23 +80,44 @@ public void setUp() {
 		checkContextFields(resource);
 		jaxRsResource = spy(resource);
 
-		ConfigurationWarnings globalConfigWarnings = ConfigurationWarnings.getInstance();
-		globalConfigWarnings.clear();
-
 		MockServletContext servletContext = new MockServletContext();
 		MockServletConfig servletConfig = new MockServletConfig(servletContext, "JAX-RS-MockDispatcher");
 		jaxRsResource.servletConfig = servletConfig;
 		IbisContext ibisContext = mock(IbisContext.class);
-		IbisManager ibisManager = new MockIbisManager();
+		Configuration configuration = new TestConfiguration();
+		IbisManager ibisManager = configuration.getIbisManager();
 		ibisManager.setIbisContext(ibisContext);
 		MessageKeeper messageKeeper = new MessageKeeper();
 		doReturn(messageKeeper).when(ibisContext).getMessageKeeper();
 		doReturn(ibisManager).when(ibisContext).getIbisManager();
 		doReturn(ibisContext).when(jaxRsResource).getIbisContext();
+		doReturn(configuration.getBean("applicationWarnings")).when(ibisContext).getBean(eq("applicationWarnings"), any());
+		registerAdapter(configuration);
 
 		dispatcher.register(jaxRsResource);
 	}
 
+	private void registerAdapter(Configuration configuration) {
+		Adapter adapter = new Adapter();
+		adapter.setName("dummyAdapter");
+		try {
+			PipeLine pipeline = new PipeLine();
+			PipeLineExit exit = new PipeLineExit();
+			exit.setPath("EXIT");
+			exit.setState("success");
+			pipeline.registerPipeLineExit(exit);
+			EchoPipe pipe = new EchoPipe();
+			pipe.setName("myPipe");
+			pipeline.addPipe(pipe);
```

Comment:
Waarom moet deze pipe er zijn? 
Kan hij een andere naam krijgen, die je een hint geeft waar hij voor is?

Reply:
De pipe is er om de api (beter) te kunnen testen. Het gaat meer om de adapter dan de pipe